### ISSUE TITLE
conflict while using 'state', 'scope' or 'users' attribute in requests

### ISSUE THREAD

**Issue by ballinette** [kuzzleio/kuzzle#379]
If we try to create a document (using action write/createOrReplace, for example) with following format: 

```
{"name":"Test Univers","env":"dev","connectors":[],"status":"active","id":"","users":[],"lastUpdate":1472594348,"createDate":1472594348}
```

we have following error:

```
TypeError: additionalData[attr].toLowerCase is not a function
<br> &nbsp; &nbsp;at /home/kuzzle/kuzzle-master/node_modules/kuzzle-common-objects/models/requestObject.js:49:125
```

**Comment by ballinette** [comment]
reproduced only with REST API, not with Websocket

**Comment by dbengsch** [comment]
With REST API (didn't check with Websocket), following request will fail due to the body field :
```json
curl -x '' -X PUT -H "Content-Type: application/json" http://localhost:7511/api/1.0/example/blogpost/42/_createOrReplace -d '{
  "author": "John Doe",
  "title": "I love cats",
  "body": "They are so cute",
  "tags": [ "pet", "animal", "cat" ],
  "status": "pending",
  "publish_date": "2016-08-03"
}'
```
With error:
```json
HTTP/1.1 400 Bad Request
Access-Control-Allow-Headers: X-Requested-With
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Type: application/json
Date: Mon, 24 Oct 2016 12:45:07 GMT
Transfer-Encoding: chunked

{
    "action": "createOrReplace", 
    "collection": "blogpost", 
    "controller": "write", 
    "error": {
        "_source": {
            "_id": "42", 
            "body": "They are so cute"
        }, 
        "message": "failed to parse"
    }, 
    "index": "example", 
    "metadata": {}, 
    "requestId": "a3d4173f-08e6-4124-81a6-1686ba092bf0", 
    "result": null, 
    "status": 400
}
```

**Comment by ballinette** [comment]
will be fixed by #559
